pennylane/interfaces/tensorflow.py

```diff
@@ -94,14 +94,14 @@ def _to_tensors(x):
     return tf.convert_to_tensor(x)
 
 
-def _res_restructured(res, tapes, legacy_shots: Shots):
+def _res_restructured(res, tapes):
     """
     Reconstruct the nested tuple structure of the output of a list of tapes
     """
     start = 0
     res_nested = []
     for tape in tapes:
-        tape_shots = legacy_shots or tape.shots or Shots(1)
+        tape_shots = tape.shots or Shots(1)

```

Comment:
This seems fine because it is unrelated, but I was actually surprised to see this single-shot logic here for cases like `tape.shots=None`.